src/communicate.rs

```diff
@@ -143,7 +149,7 @@ pub fn write_to_peer_connection(peer_addr: SocketAddr, conn: &QConn, wire_msg: W
                 )
             })
         })
-        .map(|_| ());
+        .map(move |_| utils::handle_send_success(peer_addr, user_msg2));
```

Comment:
Will that do the right thing?
i.e will we not fail to send the success message if `tokio::io::write_all` return None? or tokio shutdown.

Reply:
My bad I guess, in these case it will call handle_communication_err